### PR TITLE
trigger workflow on pull_request

### PR DESCRIPTION
**Pull request by finn-tbd**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 name: Tests
-on: [push]
+on:
+  push:
+  pull_request: {}
```
Comment (leordev) [inline .github/workflows/tests.yml:4]: qq: is `{}` really needed?

**Review by leordev — APPROVED** [review]
lgtm

**Review by leordev — CHANGES_REQUESTED** [review]
Actually the pull request with no filters will duplicate the run with the push with no branch filters. So the CI will run twice. 

To avoid that we normally leave pull_request with no filters and push only filtering the main branch.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (finn-tbd) [reply .github/workflows/tests.yml]: depends on the yaml parser! I think github doesnt require it, so i've removed.

**Comment by finn-tbd** [comment]
oh that's dumb. Fixed, although I dislike the requirement to open a PR before you can get CI checks.

**Comment by leordev** [comment]
> oh that's dumb. Fixed, although I dislike the requirement to open a PR before you can get CI checks.

Agree, I also find that annoying... I usually overcome by adding a manual trigger (`workflow_dispatch`). And using my `gh` cli to test it upfront. Feel free to add it if you like that workaround.

**Review by leordev — APPROVED** [review]

**Pull request merged**